### PR TITLE
Proper handling of Wheels and Flight Sticks in FreeFlyController.

### PR DESCRIPTION
**Pull request by gleblebedev**
AxisAdapter transforms input value taking into account neutral value, dead zone width and sensitivity. Then the value been used in FreeFlyController and DirectionalInput

### CODE REVIEW

File: Source/ThirdParty/SDL/src/joystick/SDL_joystick.c [hunk 1]
```diff
@@ -2309,9 +2309,11 @@ static SDL_bool SDL_IsJoystickProductArcadeStick(Uint32 vidpid)
 static SDL_bool SDL_IsJoystickProductFlightStick(Uint32 vidpid)
 {
     static Uint32 flightstick_joysticks[] = {
-        MAKE_VIDPID(0x044f, 0x0402),    /* HOTAS Warthog Joystick */
         MAKE_VIDPID(0x0738, 0x2221),    /* Saitek Pro Flight X-56 Rhino Stick */
```
Comment (eugeneko) [inline Source/ThirdParty/SDL/src/joystick/SDL_joystick.c:2312]: Please push these changes in SDL repo
https://github.com/rbfx/SDL/tree/rbfx

File: Source/Urho3D/Input/FreeFlyController.cpp [hunk 2]
```diff
@@ -31,17 +31,15 @@
...
 namespace
 {
-
-float ApplyDeadZone(float value, float deadZone)
+float ClampPitch(float pitch)
 {
-    if (value >= -deadZone && value <= deadZone)
-        return 0.0f;
-    return (value - Sign(value) * deadZone) / (1.0f - deadZone);
+    return Clamp(pitch, -90.0f, 90.0f);
```
Comment (eugeneko) [inline Source/Urho3D/Input/FreeFlyController.cpp:42]: There may be a problem with high pitches. Let's make this constant configurable.

File: Source/Urho3D/Input/AxisAdapter.cpp [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+    if (value > neutral_)
+    {
+        const float rangeMin = neutral_ + deadZone_;
+        const float srcRange = 1.0f - rangeMin;
+        const float normalized = (value - rangeMin) / srcRange;
+        const float exp = GetExponent(posSensitivity_);
+        const float transformed = Pow(normalized, exp);
+        return transformed;
+    }
+    {
```
Comment (eugeneko) [inline Source/Urho3D/Input/AxisAdapter.cpp:95]: wat
"else" missing?

**Review by eugeneko — DISMISSED** [review]

File: Source/Urho3D/Input/AxisAdapter.cpp (hunk 3 above)
Comment (gleblebedev) [reply Source/Urho3D/Input/AxisAdapter.cpp:95]: On purpose because of return statement.

File: Source/Urho3D/Input/FreeFlyController.cpp (hunk 2 above)
Comment (gleblebedev) [reply Source/Urho3D/Input/FreeFlyController.cpp:42]: I've fixed gimbal lock with one of previous PRs but if you still concerned - sure.

**Comment by eugeneko** [comment]
It’s not about gimbal lock, is about precision loss. Try sample-project
with your changes, and try to run forward when looking down.

**Review by eugeneko — APPROVED** [review]

**Pull request merged**